### PR TITLE
Added Snatchpad keyboard

### PR DESCRIPTION
**Pull request by xia0**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/snatchpad/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        tap_code16(LCTL(KC_Z));
+      }
+        return false;
+
+
+
+    }
+  }
+  return false;
+}
```
Comment (drashna) [inline keyboards/handwired/snatchpad/keymaps/default/keymap.c:90]: This code should be moved to the keyboard's c file. 
```suggestion
```

File: keyboards/handwired/snatchpad/readme.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# snatchpad
+
+![snatchpad etch-a-sketch macropad](https://github.com/xia0/keeb_files/blob/main/snatchpad/images/20220213_0004.jpg?raw=true)
```
Comment (drashna) [inline keyboards/handwired/snatchpad/readme.md:3]: please rehost the file on something like Imgur, and link that here.

File: keyboards/handwired/snatchpad/readme.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+A 6-key macropad with 2 rotary encoders intended for use as a pointing device. [Vial](https://github.com/vial-kb/vial-qmk) can be used to remap the encoders.
```
Comment (drashna) [inline keyboards/handwired/snatchpad/readme.md:5]: Please remove all references to vial, as it is not officially supported by QMK

```suggestion
A 6-key macropad with 2 rotary encoders intended for use as a pointing device. 
```

File: keyboards/handwired/snatchpad/snatchpad.c [hunk 4]
```diff
@@ -0,0 +1,8 @@
+// Copyright 2022 xia0 (@xia0)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "snatchpad.h"
+
+bool encoder_update_kb(uint8_t index, bool clockwise) {
+    return encoder_update_user(index, clockwise);
+}
```
Comment (drashna) [inline keyboards/handwired/snatchpad/snatchpad.c:8]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) {
        return false;
    }

    uint8_t layer = get_highest_layer(layer_state);

    if (index == 0) { /* First encoder */
        switch (layer) {
            case 0:
                if (clockwise) {
                    tap_code_delay(KC_VOLU, 10);
                } else {
                    tap_code_delay(KC_VOLD, 10);
                }
                break;
            case 2:
                if (clockwise) {
                    tap_code16(LCTL(KC_MINUS));
                } else {
                    tap_code16(LCTL(KC_EQUAL));
                }
                break;
            default:
                if (clockwise) {
                    tap_code(KC_MS_L);
                } else {
                    tap_code(KC_MS_R);
                }
                break;
        }

    } else if (index == 1) { /* Second encoder */
        switch (layer) {
            case 0:
                if (clockwise) {
                    tap_code(KC_MFFD);
                } else {
                    tap_code(KC_MRWD);
                }
                break;
            case 2:
                if (clockwise) {
                    tap_code16(LCTL(KC_Y));
                } else {
                    tap_code16(LCTL(KC_Z));
                }
                break;
            default:
                if (clockwise) {
                    tap_code(KC_MS_D);
                } else {
                    tap_code(KC_MS_U);
                }
                break;
        }
    }
    return true;
}
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/snatchpad/readme.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
+# snatchpad
+
+![snatchpad etch-a-sketch macropad](https://i.imgur.com/aRIM8Na.jpg)
```
Comment (fauxpark) [inline keyboards/handwired/snatchpad/readme.md:3]:
```suggestion
![snatchpad etch-a-sketch macropad](https://i.imgur.com/aRIM8Nah.jpg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**